### PR TITLE
Add the linter and update docs.

### PR DESCRIPTION
**Pull request by hoch**
No functional changes.

- Created package.json for node modules.
- Added packages: ESLint, ESLint Google JS style and ESLint HTML plugin.
- Applied the linter to the `design-pattern/awn` example. (As a pilot)
- Added CONTRIBUTING and fixed README accordingly.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,27 +1,202 @@
...
-// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
```
Comment (rtoy) [inline LICENSE]: Are we actually allowed to change the license terms?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+Follow either of the two links above to access the appropriate CLA and instructions for how to sign and return it. Once we receive it, we'll be able to accept your pull requests.
+
+## Contributing A Patch
+
+1. Submit an issue describing your proposed change to the repo in question.
+1. The repo owner will respond to your issue promptly.
+1. If your proposed change is accepted, and you haven't already done so, sign a Contributor License Agreement (see details above).
+1. Fork the desired repo, develop and test your code changes.
+1. Ensure that your code adheres to the existing style in the sample to which you are contributing.
+1. Submit a pull request.
```
Comment (rtoy) [inline CONTRIBUTING.md]: updating docs and adding linter seem totally unrelated.  But oh well.

File: README.md [hunk 3]
```diff
@@ -7,3 +7,21 @@ This is a collection of Web Audio API examples.
  - [Stress Test](http://googlechromelabs.github.io/web-audio-samples/stress-test/boxes)
 
 __NOTE__: Some of these samples are not functional or cross-browser compatible. Please feel free to file issues or submit PRs.
+
+
+## Support
+
+If you have found an error in this library, please file an issue at: https://github.com/GoogleChromeLabs/web-audio-samples/issues.
+
+Patches are encouraged, and may be submitted by forking this project and submitting a pull request through GitHub. See CONTRIBUTING for more detail.
```
Comment (rtoy) [inline README.md]: Wrap lines?  Does "CONTRIBUTING" link to the doc?

File: audio-worklet/design-pattern/awn/index.html [hunk 4]
```diff
@@ -80,7 +83,7 @@ <h1 id="title-header"></h1>
           function. See the article on the AudioWorklet design pattern for more
           details.`,
       htmlSource: 'audio-worklet/design-pattern/awn/',
-      demoFunction: runDemo
+      demoFunction: runDemo,
```
Comment (rtoy) [inline audio-worklet/design-pattern/awn/index.html]: Did eslinter actually add the trailing comma?  It's not really wrong not to have it, is it?

File: audio-worklet/design-pattern/lib/wa2.js [hunk 5]
```diff
@@ -113,16 +113,17 @@ class HeapAudioBuffer {
...
    * data.
    *
    * @param  {number|undefined} channelIndex Channel index.
-   * @return {Float32Array|Array[Float32Array]} a channel data array or an
+   * @return {?Array} a channel data array or an
    * array of channel data.
    */
   getChannelData(channelIndex) {
-    if (channelIndex < this._channelCount)
+    if (channelIndex >= this._channelCount) {
```
Comment (rtoy) [inline audio-worklet/design-pattern/lib/wa2.js]: Haha. :-( Totally missed that in the first review.

File: LICENSE (hunk 1 above)
Comment (hoch) [reply LICENSE]: The previous license file was not aligned with Google's open source guideline. This PR fixes it. Now it is compliant to the guideline.

File: CONTRIBUTING.md (hunk 2 above)
Comment (hoch) [reply CONTRIBUTING.md]: I just put all the non-functional change into this PR. I can break this down to multiple PRs if that's helpful.

File: README.md (hunk 3 above)
Comment (hoch) [reply README.md]: This is a template text from other Google open-source project. (e.g. Omnitone) I would like to keep it as is.

File: audio-worklet/design-pattern/awn/index.html (hunk 4 above)
Comment (hoch) [reply audio-worklet/design-pattern/awn/index.html]: Yes. Surprisingly this is recommended/enforced by Google ESlint plugin. I kinda like it because the change can be simpler after this treatment.

File: audio-worklet/design-pattern/lib/wa2.js (hunk 5 above)
Comment (hoch) [reply audio-worklet/design-pattern/lib/wa2.js]: Heh. Let's fix it now.
